azurerm/resource_arm_cognitive_account.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmCognitiveAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmCognitiveAccountCreate,
+		Read:   resourceArmCognitiveAccountRead,
+		Update: resourceArmCognitiveAccountUpdate,
+		Delete: resourceArmCognitiveAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kind": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCognitiveAccountKind,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validateCognitiveAccountSkuName,
+						},
+
+						"tier": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validateCognitiveAccountSkuTier,
+						},
+					},
+				},
+				Set: resourceArmCognitiveAccountSkuHash,
+			},
+			"location":            locationSchema(),
+			"tags":                tagsSchema(),
+			"resource_group_name": resourceGroupNameSchema(),
+		},
+	}
+}
+
+func resourceArmCognitiveAccountSkuHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%v-", m["name"]))
+	buf.WriteString(fmt.Sprintf("%v-", m["tier"]))
+
+	return hashcode.String(buf.String())
+}
+
+type dummy struct{}
+
+func resourceArmCognitiveAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cognitiveAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+	fmt.Printf("[INFO] preparing arguments for Azure ARM  Cognitive Servies Account creation.\n")
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	kind := d.Get("kind").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	fmt.Printf("[DEBUG] name: %s location: %s resGroup: %s kind: %s \n", name, location, resGroup, cognitiveservices.Kind(kind))
+	sku := expandCognitiveSku(d)
+	prop := dummy{}
+
+	createParameter := cognitiveservices.AccountCreateParameters{
+		Kind:       cognitiveservices.Kind(kind),
+		Location:   &location,
+		Sku:        &sku,
+		Properties: &prop,
+		Tags:       *expandTags(tags),
+	}
+
+	account, err := client.Create(ctx, resGroup, name, createParameter)
+	if err != nil {
+		fmt.Printf("Error: %s \n", err.Error())
+		return err
+	}
+	if account.AccountProperties.ProvisioningState != cognitiveservices.Succeeded {
+		return fmt.Errorf("Error provisioning Account")
+	}
```

Comment:
I think we're safe to remove this check (since the SDK will return an error if it's not a `200 OK`)